src/i18n.js

```diff
@@ -33,3 +33,5 @@ module.exports = async function () {
     ipcMain.emit('languageUpdated', lang)
   })
 }
+
+module.exports = setupI18n()
```

Comment:
The problem here is that this dialog _might_ be the first action IPFS Desktop will take. When you open an `ipfs://` URL on your computer and IPFS Desktop is shut down, it will start up and show this dialog. If the i18n isn't loaded, it won't work. However, I saw your PR and yes, there are better ways to do it. Perhaps we can merge #2114 before and make sure to rebase this correctly to ensure the i18n stuff is loaded before any window is open.

Reply:
I hope to get the ctx refactor done late today if my ESM migration of js-pinning-service-http-client can finish today.

I do want to clarify that even if the dialog is required as the first time IPFS desktop is opened, we can keep the prior function as is, but cache the result (when appropriate) and await any pre-reqs inside the place where we need it; not globally. In general, we should

1. Make everything async. The refactor ctx PR migrates many things to async, but I'm not as comfortable with the app as I should be so I tried not to change things unnecessarily.
2. Execute on first use, not on require/import (whether in src/index or src/dialogs/prompt/index), caching the resulting value if re-evaluations are not necessary. `if (result != null) return result` at the top of everything.
3. Await each required dependency & pre-req within each module. 

With these three simple rules, without even trying, anything/everything we need is 99% of the way to being [SOLID](https://en.wikipedia.org/wiki/SOLID). This is essentially the same as dynamic imports, but in the form of asynchronously awaiting required dependencies & pre-req processes.


https://github.com/ipfs/ipfs-desktop/pull/1966#discussion_r866967386 mentions more about doing this.